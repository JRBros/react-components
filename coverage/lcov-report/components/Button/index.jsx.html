<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/Button/index.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">components/Button</a> index.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/4</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import { PropTypes } from 'react';
import styled from 'styled-components';
&nbsp;
import { GRAY, GRAY_LIGHT } from '../../constants/colors';
&nbsp;
const Button = styled.button`
    /* Box model */
    width: ${props =&gt; (props.fullWidth ? <span class="branch-0 cbranch-no" title="branch not covered" >'100%' </span>: props.width)}
    padding: .6rem 1rem;
&nbsp;
    /* Typo */
    color: ${GRAY};
&nbsp;
    /* Visual */
    background-color: ${GRAY_LIGHT};
    border: none;
    outline: none;
`;
&nbsp;
Button.defaultProps = {
    fullWidth: false,
    width: 'auto'
};
&nbsp;
Button.propTypes = {
    fullWidth: PropTypes.bool,
    width: PropTypes.string
};
&nbsp;
export default Button;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="http://istanbul-js.org/" target="_blank">istanbul</a> at Sat Mar 18 2017 19:08:06 GMT+0100 (CET)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
