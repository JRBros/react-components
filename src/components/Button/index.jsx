import { PropTypes } from 'react';
import styled from 'styled-components';

import { GRAY, GRAY_LIGHT } from '../../constants/colors';

const Button = styled.button`
    /* Box model */
    width: ${props => (props.fullWidth ? '100%' : props.width)}
    padding: .6rem 1rem;

    /* Typo */
    color: ${GRAY};

    /* Visual */
    background-color: ${GRAY_LIGHT};
    border: none;
    outline: none;
`;

Button.defaultProps = {
    fullWidth: false,
    width: 'auto'
};

Button.propTypes = {
    fullWidth: PropTypes.bool,
    width: PropTypes.string
};

export default Button;
