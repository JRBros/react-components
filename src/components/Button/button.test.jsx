import React from 'react';
import renderer from 'react-test-renderer';

import Button from './index.jsx';

test('Button', () => {
    const component = renderer.create(
       <Button>Default Button</Button>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
