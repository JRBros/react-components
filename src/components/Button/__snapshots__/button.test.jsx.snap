// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button 1`] = `
<button
  className="cHzpQi"
  width="auto"
>
  Default Button
</button>
`;
